frontend/src/main/pages/PersonalSchedules/PersonalSchedulesWeeklyViewPage.js

```diff
@@ -0,0 +1,129 @@
+import React from "react";
+import { useParams, useNavigate } from "react-router-dom";
+import BasicLayout from "main/layouts/BasicLayout/BasicLayout";
+import PersonalSchedulePanel from "main/components/PersonalSchedules/PersonalSchedulePanel";
+import { useBackend } from "main/utils/useBackend";
+import { Button, Row, Col } from "react-bootstrap";
+
+// Helper function to convert HH:MM (24-hour) to h:MM AM/PM format
+// Stryker disable all
+const formatTime = (timeString) => {
+  if (!timeString) return "";
+  const [hours, minutes] = timeString.split(":");
+  const hoursInt = parseInt(hours, 10);
+  const ampm = hoursInt >= 12 ? "PM" : "AM";
+  const formattedHours = hoursInt % 12 || 12; // Convert 0 or 12 to 12
+  return `${formattedHours}:${minutes} ${ampm}`;
+};
```

Comment:
Can we put this into a utility file so that it can be reused rather than being copy/pasted into multiple components?

See, for example: 

https://github.com/ucsb-cs156-s25/proj-courses-s25-02/blob/main/frontend/src/main/utils/timeUtils.js

where you will find very similar code.. so similar you might even be able reuse it!

It is also already covered by tests... when you put this into a separate file it is much easier to cover it with tests, because you only have to do it once.